packages/template-retail-react-app/app/contexts/index.js

```diff
@@ -5,8 +5,10 @@
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
-import React, {useState} from 'react'
+import React, {useEffect, useState} from 'react'
 import PropTypes from 'prop-types'
+import isEqual from 'lodash/isEqual'
```

Comment:
The bloat I mean is the unused functionality. Most of what lodash does can now be done natively. In fact, the only thing we currently use lodash for is debounce, which we could also easily remove. And even if we don't import the code, we still take the hit of installing it, when we could use other, smaller packages (e.g. [fast-deep-equal](https://www.npmjs.com/package/fast-deep-equal)). And there's the maintenance burden. Lodash is notoriously slow to update, and for a while had multiple open vulnerabilities that just caused toil for us.

Reply:
So when you say "Most of what lodash does can now be done natively" what is the native method for deep object equality?

Reply:
There's no native method for a deep equality check, but the implementation for our use case (pure JSON objects) is fairly simple. I don't think it's worth the file size or maintenance burden of including lodash, when other options are available.

Reply:
Two strings of valid JSON can easily fail an object equality test. All you have to do is swap the order of keys. I'm fine with `fast-deep-equal` but I also feel we're getting a bit pedantic here. This decision was made simply because we already import `lodash` (hence, have gotten security signoff and gone through that whole circus) so it was used here